rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/is_hvac_sys_preheat_fluid_loop_attached_to_boiler.py

```diff
@@ -0,0 +1,56 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all, find_one_with_field_value
+
+FLUID_LOOP = schema_enums["FluidLoopOptions"]
+
+
+def is_hvac_sys_preheat_fluid_loop_attached_to_boiler(rmi_b, hvac_b_id):
+    """Returns True if the fluid loop associated with preheat system associated with the HVAC system is attached to a boiler.
+    Returns False if this is not the case.
+
+    Parameters
+    ----------
+    rmi_b : json
+        RMD at RuleSetModelInstance level
+    hvac_b_id : str
+        The HVAC system ID.
+
+    Returns
+    -------
+    bool
+        True: preheat system is attached to a boiler
+        False
+    """
+    is_hvac_sys_preheat_fluid_loop_attached_to_boiler_flag = False
+    boilers = find_all("$.boilers[*]", rmi_b)
+    loop_boiler_dict = dict()
+    for boiler_b in boilers:
+        loop_id = getattr_(boiler_b, "boiler", "loop")
+        if not loop_id in loop_boiler_dict.keys():
+            loop_boiler_dict[loop_id] = []
+        loop_boiler_dict[loop_id].append(boiler_b)
+
+    # Get the hvac system
+    hvac_b = find_one_with_field_value(
+        "$.buildings[*].building_segments[*].heating_ventilation_air_conditioning_systems",
+        "id",
+        hvac_b_id,
+        rmi_b,
+    )
+    # Check if hvac_b has preheat system
+    if hvac_b and hvac_b.get("preheat_system"):
+        hot_water_loop_id = hvac_b["preheat_system"].get("hot_water_loop")
+        # Check if the preheat system has hot water loop and the hot water loop connects to a boiler(s)
+        if hot_water_loop_id and hot_water_loop_id in loop_boiler_dict.keys():
```

Comment:
In ```if hot_water_loop_id and hot_water_loop_id```, Was two ```hot_water_loop_id```s intended? Or was it to be ```if hot_water_loop_id in loop_boiler_dict.keys():```?

Reply:
The first `if hot_water_loop_id` is to prevent `None` id.